onnxruntime/server/http/core/session.cc

```diff
@@ -95,10 +98,21 @@ template <typename Body, typename Allocator>
 void HttpSession::HandleRequest(http::request<Body, http::basic_fields<Allocator> >&& req) {
   HttpContext context{};
   context.request = std::move(req);
+  // Log the requests path to save double handling and monitor
+  const auto path = std::string(context.request.target().to_string());
+  // Record how many total requests have been processed by this server
+  // this can be used to get QPS metrics
+  MetricRegistry::get().totalHTTPRequests->Add(
+      {{"component", "http"}, {"path", path}}).Increment();
 
   // Special handle the liveness probe endpoint for orchestration systems like Kubernetes.
-  if (context.request.method() == http::verb::get && context.request.target().to_string() == "/") {
+  if (context.request.method() == http::verb::get && path == "/") {
     context.response.body() = "Healthy";
+  } else if (context.request.method() == http::verb::get && path == "/metrics") {
```

Comment:
I think the best way to do this would be to create a `RegisterGet` method in `http_server.cc` and then make a GET callback in main.